lona/client/window.js

```diff
@@ -30,14 +30,19 @@ Lona.LonaWindow = function(lona_context, root, window_id) {
     this._crashed = false;
     this._view_running = false;
     this._view_runtime_id = undefined;
-    this._url = '';
+    this._url = undefined;
     this._nodes = {};
     this._widget_marker = {};
     this._widget_data = {};
     this._widgets = {};
     this._widgets_to_setup = [];
     this._widgets_to_update = [];
 
+    // helper -----------------------------------------------------------------
+    this.get_url = function() {
+        return this._url.pathname + this._url.search;
```

Comment:
Don't we need to use `this._url.hash` as well?

Reply:
Good point! Fixed